// ngx-tinymce/esm2022/src/tinymce.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TinymceOptions } from './tinymce.options';
import { TinymceComponent } from './tinymce.component';
import * as i0 from "@angular/core";
export class NgxTinymceModule {
    static forRoot(options) {
        return {
            ngModule: NgxTinymceModule,
            providers: [{ provide: TinymceOptions, useValue: options }],
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.3", ngImport: i0, type: NgxTinymceModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "18.0.3", ngImport: i0, type: NgxTinymceModule, imports: [CommonModule, TinymceComponent], exports: [TinymceComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "18.0.3", ngImport: i0, type: NgxTinymceModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.3", ngImport: i0, type: NgxTinymceModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, TinymceComponent],
                    exports: [TinymceComponent],
                }]
        }] });
